libslide/src/bignum/sub.rs

```diff
@@ -0,0 +1,148 @@
+#![allow(clippy::suspicious_arithmetic_impl)]
+use crate::bignum::utils::is_smaller;
+use crate::bignum::Bignum;
+use std::convert::TryInto;
+use std::mem;
+use std::ops;
+
+impl ops::Sub for Bignum {
+    type Output = Bignum;
+
+    fn sub(self, other: Bignum) -> Bignum {
+        // make the decimal vectors the same size
+        // fill lhs and rhs vectors as well
+        let mut lhs_dec: Vec<u8>;
+        let mut res_dec: Vec<u8>;
+        let self_size: usize = self.dec.len();
+        let other_size: usize = other.dec.len();
+        let (mut res_int, mut lhs_int) = if self_size > other_size {
+            res_dec = self.dec;
+            lhs_dec = vec![0; self_size];
+            lhs_dec[..other_size].clone_from_slice(&other.dec[..other_size]);
```

Comment:
I'm confused by this. Why not just
```suggestion
            lhs_dec = other.dec
```

Reply:
Got it, nevermind. Is it possible to move a slice into another? If not another option is to set `lhs_dec = other.dec`, then `.reserve()` the rest of the space needed and add 0s. This is pedantic though, the cost savings are just copying some 8-bit values and extra zeros.